src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -0,0 +1,56 @@
+window.addEventListener('load', () => {
+	/**
+	 * Handle fields visibility
+	 * @unreleased
+	 */
+	function handleVisibility(donationForm) {
+		donationForm.querySelectorAll('[data-field-visibility-conditions]').forEach(function (inputField) {
+			const fieldWrapper = inputField.closest('.form-row');
+			const visibilityConditions = JSON.parse(inputField.getAttribute('data-field-visibility-conditions'));
+			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
+			let visible = false;
+			const {field, value} = visibilityCondition;
+
+			const inputs = donationForm.querySelectorAll(`[name="${field}"]`);
+			let hasFieldController = !! inputs.length;
+
+			// Do not apply visibility conditions if field controller does not exit in DOM.
+			if ( ! hasFieldController ) {
+				return;
+			}
+
+			inputs.forEach((input) => {
+				const fieldType = input.getAttribute('type');
+
+				if (fieldType && (fieldType === 'radio' || fieldType === 'checkbox')) {
+					if (input.checked && input.value === value) {
+						visible = true;
+					}
+				} else if (input.value === value) {
+					visible = true;
+				}
+			});
+
+			// Show or Hide field wrapper.
+			visible ?
+				fieldWrapper.classList.remove('give-hidden') :
+				fieldWrapper.classList.add('give-hidden');
+		});
+	}
+
+	/**
+	 * @unreleased
+	 * @param event
+	 */
+	function handleVisibilityOnChangeHandler(event) {
+		handleVisibility(event.target.closest('.give-form') );
+	}
+
+	document.querySelectorAll('form.give-form')
+		.forEach(function (donationForm) {
+			handleVisibility(donationForm);
+			donationForm.querySelectorAll('input, select, textarea')
+				.forEach(field => field.addEventListener('change', handleVisibilityOnChangeHandler));
+		});
+});
```

Comment:
@ravinderk If I'm reading this right, this has the potential to cause a lot of overhead, especially if the user is typing quickly into an input. Unless I'm missing something this, combined with the `handleVisibility` function, would go through every single conditional field every single change (e.g. keystroke). If so, that's way too much.

Another method here would be something like:
1. When the JS first loads, grab all the conditional fields
2. Get the list of fields being watched
3. Loop through those fields and create a change callback closure
4. In the closure, adjust the target field visibility based on the current watched field's value

Also, please support at least `=` and `!=` a this point.

Reply:
@JasonTheAdams Noted.

Reply:
@JasonTheAdams logic has been updated